js/directives/lucca/user-picker.js

```diff
@@ -0,0 +1,322 @@
+				}
+				else {
+					scope.properties = DEFAULT_HOMONYMS_PROPERTIES;
+				}
+			}
+		};
+	})
+
+	.directive('luidUserPickerMultiple', function () {
+		return {
+			restrict: 'E',
+			controller: "luidUserPickerController",
+			template: userPickerMultipleTemplate,
+			require: "luidUserPickerController",
+			scope: {
+				showFormerEmployees: "=",
+				onSelect: "&",
+				onRemove: "&",
+				controlDisabled: "=",
+				useCustomFilter: "=",
+				customFilter: "&",
+				appId: "@",
+				operation: "@",
+				homonymsProperties: "@"
+			},
+			link: function (scope, elt, attrs, ctrls) {
+				if (attrs.homonymsProperties) {
+					scope.properties = attrs.homonymsProperties.split(',');
+				}
+				else {
+					scope.properties = DEFAULT_HOMONYMS_PROPERTIES;
+				}
+			}
+		};
+	})
+
+	.controller("luidUserPickerController", ['$scope', '$http', 'moment', '$timeout', '$q', function ($scope, $http, moment, $timeout, $q) {
+		var timeout = {};
+		var promise;
+		var maxUsers = 0;
+		var selectedUsersCount = 0;
+
+		$scope.user = {};
+		$scope.selected = {};
+		$scope.selected.users = [];
+
+		var getUsers = function(init, input) {
+			var formerEmployees = "formerEmployees=" + ($scope.showFormerEmployees ? "true" : "false");
+			var limit = "&limit=";
+			var clue = "&clue=" + input;
+			var query = "/api/v3/users/find?" + (init ? "" : (clue + "&")) + formerEmployees;
+
+			if (promise) {
+				promise.then(function (response) {}); // do nothing
+			}
+
+			// Cancel previous timeout
+			if (timeout.count) {
+				$timeout.cancel(timeout.count);
+			}
+
+			// If we use a UserPickerMultiple or a custom filter, we ask for all the users
+			if ((selectedUsersCount !== 0) || ($scope.useCustomFilter)) {
+				limit += MAGIC_NUMBER_maxUsers;
+			}
+			else {
+				limit += (MAX_COUNT + 1);
+			}
+			query += limit;
+
+			promise = $http.get(query);
+			promise.then(function (response) {
+				$scope.users = response.data.data.items;
+
+				// Handle pagination
+				if (!$scope.useCustomFilter && (selectedUsersCount === 0)) {
+					// If there are more results than MAX_COUNT, launch a timer to get the number of results
+					if ($scope.users.length > MAX_COUNT) {
+						countTimeout(input);
+						$scope.users = _.first($scope.users, MAX_COUNT);
+						$scope.users.push({ overflow: MAX_COUNT + "/..." });
+					}
+				}
+				else {
+					// Used for UserPickerMultiple only
+					if (selectedUsersCount > 0) {
+						// Remove duplicates between results and selected users
+						_.each($scope.selected.users, function (selectedUser) {
+							$scope.users = _.reject($scope.users, function (user) {
+								return (user.id === selectedUser.id);
+							});
+						});
+						// Number of results matching the query
+						$scope.count = $scope.users.length;
+					}
+
+					// Used when a custom filtering function is given
+					if ($scope.useCustomFilter) {
+						// Get new set of users
+						$scope.users = $scope.customFilter({ users: $scope.users });
```

Comment:
i'd use customFilter another way. i'd ask for a function with this signature 
```js
customFilter: function(user){ return boolean; } 
```
and have this line here
```js
$scope.users = _.filter(users, $scope.customFilter);
```

The question is to know what kind of custom filter the user want to pass.

In the case of Timmi we only want to dislay users from a specific population so the filter is 
```js
var customFilter = function(user){
  return isPartOfThePopulation(user);
}
```
its a condition on each user specifically, by that i mean that it is not dependent on the other users. I can't think of an example where this would be pertinent
```js
var customFilter = function(user, users){ return true/false; };
```
By that i mean that we filter a user because some other user is present